Content.Server/DeltaV/StationEvents/Events/PirateRadioSpawnRule.cs

```diff
@@ -1,32 +1,91 @@
 using Content.Server.GameTicking.Rules.Components;
 using Content.Server.StationEvents.Components;
-using Content.Server.RoundEnd;
+using Content.Server.Station.Components;
+using Content.Shared.Salvage;
+using Content.Shared.Random.Helpers;
+using System.Linq;
 
 namespace Content.Server.StationEvents.Events;
 
 public sealed class PirateRadioSpawnRule : StationEventSystem<PirateRadioSpawnRuleComponent>
 {
+    [Dependency] private readonly IEntityManager _entities = default!;
     [Dependency] private readonly IMapManager _mapManager = default!;
     [Dependency] private readonly MapLoaderSystem _map = default!;
-    [Dependency] private readonly GameTicker _gameTicker = default!;
-    [Dependency] private readonly TraitorRuleSystem _TraitorRuleSystem = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
 
     protected override void Started(EntityUid uid, PirateRadioSpawnRuleComponent component, GameRuleComponent gameRule, GameRuleStartedEvent args)
     {
+        //This handles spawning the Listening Outpost
         base.Started(uid, component, gameRule, args);
-		
-        var shuttleMap = _mapManager.CreateMap();
-        var options = new MapLoadOptions
+
+        var xformQuery = GetEntityQuery<TransformComponent>();
+        //Find where the station is and get a bounding box
+        var aabbs = EntityQuery<StationDataComponent>().SelectMany(x =>
+                x.Grids.Select(x =>
+                    xformQuery.GetComponent(x).WorldMatrix.TransformBox(_mapManager.GetGridComp(x).LocalAABB)))
+            .ToArray();
+        if (aabbs.Length < 1) return;
+        var aabb = aabbs[0];
+
+        for (var i = 1; i < aabbs.Length; i++)
+        {
+            aabb.Union(aabbs[i]);
+        }
+        //Generates a potential spawning area, shaped like a square with rounded corners.
+        //DistanceModifier allows for fine tuning of how large this area is.
+        //The initial target distance(for DistanceModifier = 20f) is between 1km and 1.5km of the station.
+        //But with the way NextVector2 works, it can be less than that
+        //Do not for any reason set DistanceModifier greater than 25f or less than 1f. 
+        var a = MathF.Max(aabb.Height / 2f, aabb.Width / 2f) * component.DistanceModifier;
+        var randomoffset = _random.NextVector2(a, a * 2.5f);
+        var outpostOptions = new MapLoadOptions
         {
-            LoadMap = true,
+            Offset = aabb.Center + randomoffset,
+            LoadMap = false,
         };
+        //Now spawn the Listening Outpost
+        _map.TryLoad(GameTicker.DefaultMap, component.PirateRadioShuttlePath, out var outpostids, outpostOptions);
+
+        //Now we generate the outpost's debris field
+        if (outpostids == null) return;
+        //Yes, this is a loop within a loop. Actually, foreach is just here to convert an array into a variable.
```

Comment:
Loop within a loop isn't *that* uncommon or inappropriate. Remove this comment line